vispy/visuals/text/text.py

```diff
@@ -474,7 +477,14 @@ def color(self):
 
     @color.setter
     def color(self, color):
-        self._color = Color(color)
+        if isinstance(color, str):
+            self._color = ColorArray(Color(color))
+        else:
+            color = np.atleast_2d(color).astype(np.float32)
+            if color.shape[1] not in [3, 4]:
```

Comment:
Doesn't ColorArray handle all of these cases? `str` and not having elements with length `3` or `4`?

Reply:
indeed it does! thanks. fixed in d7057b6a4490eacb7d8d2b04e890b73a47b8b3f3